contracts/price-providers/V2TokenCollateralConvertPriceProviderFallback.sol

```diff
@@ -0,0 +1,71 @@
+pragma solidity 0.5.8;
+pragma experimental ABIEncoderV2;
+
+import "openzeppelin-eth/contracts/math/SafeMath.sol";
+import "openzeppelin-eth/contracts/math/Math.sol";
+
+import "./PriceProviderFallback.sol";
+import "../interface/IMoCV2.sol";
+import "../interface/IPriceProvider.sol";
+
+/**
+  @notice gets the technical price from V2 Protocol and convert to
+  @dev Parameters the MoC V2 address
+  The TP address is need to verified if is the protocol has valid price or not
+*/
+contract V2TokenCollateralConvertPriceProviderFallback is PriceProviderFallback {
+
+  using SafeMath for uint256;
+  uint256 public constant RATE_PRECISION = uint256(10**18);
+
+  IMoCV2 public mocV2;
+  address public tpAddress;
+  address public caPriceProviderAddress;
+
+  constructor(
+    IMoCV2 _mocV2,
+    address _tpAddress,
+    address _caPriceProviderAddress,
+    IMoCDecentralizedExchange _dex,
+    address _baseToken,
+    address _secondaryToken
+  ) public PriceProviderFallback(_dex, _baseToken, _secondaryToken) {
+    mocV2 = _mocV2;
+    tpAddress = _tpAddress;
+    caPriceProviderAddress = _caPriceProviderAddress;
```

Comment:
This could have been taken from the bucket itself, right? isn't it better to avoid setup error?

Reply:
More important, if for any reason the PP is changed on the bucket, this will be pointing to a potentially deprecated PP.

Reply:
we are not going to use this contract